### PR TITLE
Added option for setting the tile size (default is 256 Pixel per Tile)

### PR DESCRIPTION
**Pull request by jlaba**
Add the option tilesize in gdal2tiles.py to specify the tile size in Pixel / Tile on the command line.

### CODE REVIEW

**Comment by rouault** [comment]
Can you also update gdal/doc/source/programs/gdal2tiles.rst ?

**Comment by jlaba** [comment]
Sure no problem. I have added the option to the rst file. Hope the added version is correct.

File: gdal/doc/source/programs/gdal2tiles.rst [hunk 1]
```diff
@@ -87,10 +88,16 @@ can publish a picture without proper georeferencing too.
 
   .. versionadded:: 2.3
 
+.. option:: --tilesize=<TILESIZE>
+
+  Pixel size of the tiles.
+
+  .. versionadded:: 3.0.5
```
Comment (rouault) [inline gdal/doc/source/programs/gdal2tiles.rst]: Being a new feature, this will be:
```suggestion
  .. versionadded:: 3.1
```

Comment (jlaba) [reply gdal/doc/source/programs/gdal2tiles.rst]: Changed it to 3.1 (Sorry I'm not aware of the Commit suggestion Feature)

**Pull request merged**